File: chapter02/src/main/java/movie/Movie.java
package movie;

import money.Money;

import java.time.Duration;

public class Movie {
    private String title;
    private Duration runningTime;
    private Money fee;
    private DiscountPolicy discountPolicy;

    public Money getFee() {
        return null;
    }

    public Money calculateMovieFee(Screening screening) {
        return fee.minus(discountPolicy.calculateDiscountAmount(screening));
    }
}

File: chapter02k/src/main/kotlin/Movie.kt
import java.time.Duration

class Movie(private val title: String, private val runningTime: Duration, val fee: Money, private val discountPolicy: DiscountPolicy) {

    fun calculateMovieFee(screening: Screening): Money {
        return fee - discountPolicy.calculateDiscountAmount(screening)
    }

}

